sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/BooleanLiteralUnnecessary.vb

```diff
@@ -0,0 +1,194 @@
+﻿Imports System.Collections.Generic
+Imports System.Linq
+
+Public Class BooleanLiteralUnnecessary
+
+  Public Sub New(ByVal a As Boolean, ByVal b As Boolean, ByVal c As Boolean?)
+    Dim condition = False
+    Dim exp = True
+    Dim exp2 = True
+
+    Dim z = True OrElse ((True)) ' Noncompliant (also S1764)
+    z = False OrElse False ' Noncompliant (also S1764)
+    z = False AndAlso False ' Noncompliant (also S1764)
+    z = True AndAlso True ' Noncompliant (also S1764)
+    z = True = True ' Noncompliant (also S1764)
+    z = False = False ' Noncompliant (also S1764)
+    z = True <> True ' Noncompliant (also S1764)
+    z = False <> False ' Noncompliant (also S1764)
+    SomeFunc(True OrElse True) ' Noncompliant (also S1764)
+    Dim booleanVariable = If(condition, True, True) ' compliant (also S3923)
+
+    z = True OrElse False ' Noncompliant {{Remove the unnecessary Boolean literal(s).}}
+'            ^^^^^^^^^^^^
+    z = False OrElse True ' Noncompliant
+    z = False AndAlso True ' Noncompliant
+    z = True AndAlso False ' Noncompliant
+    z = False = True ' Noncompliant
+'             ^^^^^^
+    z = True = False ' Noncompliant
+'       ^^^^^^
+    z = False <> True ' Noncompliant
+    z = True <> False ' Noncompliant
+    Dim x = Not True ' Noncompliant
+'               ^^^^
+    x = True OrElse False ' Noncompliant
+    x = Not False ' Noncompliant
+    x = Not True ' Noncompliant
+    x = (Not ((True))) ' Noncompliant
+    x = (a = False) AndAlso True
+'                   ^^^^^^^^^^^^
+'            ^^^^^@-1
+    x = a = True ' Noncompliant
+    x = a <> False ' Noncompliant
+    x = a <> True ' Noncompliant
+    x = False = a ' Noncompliant
+'       ^^^^^
+    x = True = a ' Noncompliant
+    x = False <> a ' Noncompliant
+'       ^^^^^^^^
+    x = True <> a ' Noncompliant
+'       ^^^^
+    x = False AndAlso Foo() ' Noncompliant
+'             ^^^^^^^^^^^^^
+    x = Foo() AndAlso False ' Noncompliant
+'       ^^^^^^^^^^^^^
+    x = True AndAlso Foo() ' Noncompliant
+'       ^^^^^^^^^^^^
+    x = Foo() AndAlso True ' Noncompliant
+'             ^^^^^^^^^^^^
+    x = Foo() OrElse False ' Noncompliant
+'             ^^^^^^^^^^^^
+    x = False OrElse Foo() ' Noncompliant
+'       ^^^^^^^^^^^^
+    x = Foo() OrElse True ' Noncompliant
+'       ^^^^^^^^^^^^
+    x = True OrElse Foo() ' Noncompliant
+'            ^^^^^^^^^^^^
+    x = a = True = b ' Noncompliant
+'         ^^^^^^
+
+    x = a = Foo2(((True))) ' compliant
+    x = Not a ' compliant
+    x = Foo() AndAlso Bar() ' compliant
+    booleanVariable = If(condition, ((True)), exp) ' Noncompliant
```

Comment:
Does this compile? I'd have expected the `Then ...` part to be mandatory.

Reply:
the file compiles. The `((True))` is the "then" part